notify/driver.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2016: Mirantis Inc.
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import importlib
+import logging
+
+import jsonschema
+
+
+DRIVERS = {}
+
+
+def get_driver(name, conf):
+    """Get driver by name.
+
+    :param name: driver name
+    :param conf: dict, driver configuration
+    :rtype: Driver
+    :raises: RuntimeError
+    """
+    global DRIVERS
+    if name not in DRIVERS:
```

Comment:
there is no need in caching here. Imported modules are already cached, you won't win anything.